src/generator.rs

```diff
@@ -58,21 +57,24 @@ pub fn run_generator(devices: &[Device], output_directory: &Path, fake_mode: boo
         return Ok(());
     }
 
-    let devices_made: Vec<_> = Result::from_iter(
-        devices
-            .iter()
-            .map(|dev| handle_device(output_directory, dev)),
-    )?;
-    if !devices_made.is_empty() && !fake_mode {
-        /* We created some devices, let's make sure the module is loaded and they exist */
+    for device in devices {
+        handle_device(output_directory, device)?;
+    }
+
+    if !devices.is_empty() && !fake_mode {
+        /* We created some units, let's make sure the module is loaded and the devices exist */
         if !Path::new("/sys/class/zram-control").exists() {
             Command::new("modprobe")
                 .arg("zram")
                 .status()
                 .context("modprobe call failed")?;
         }
 
-        let max_device = devices_made.into_iter().fold(0, cmp::max);
+        let max_device = devices
+            .iter()
+            .map(|device| device.name[4..].parse().unwrap())
```

Comment:
This wants to host the `.with_context(|| format!("zram device \"{}\" number", device.name))` from `handle_device()`, AFAICT nothing prevents this same panic here as vs there.

Reply:
Hmm, but we already parsed it in `read_devices()`. So if it didn't fail there, it cannot fail here, so a panic is "good enough"?

Reply:
Ah yeah, forgot about that bit. In that case it's customary to do `.expect("already verified in read_devices()")` or something along those lines, for the benefit of both the casual reader and end-user.

Reply:
Updated like that.